pandas/tools/merge.py

```diff
@@ -1626,7 +1626,11 @@ def __init__(self, objs, axis=0, join='outer', join_axes=None,
                 clean_objs.append(v)
             objs = clean_objs
             name = getattr(keys, 'name', None)
-            keys = Index(clean_keys, name=name)
+            # GH 14848
```

Comment:
@jreback Because in the original issue, if `keys` is an `Index` containing all tuples, and `clean_keys` then becomes a list of tuples, and `name` is a single string, that's where things go wrong. What the fix does is avoid creating an `Index` (and let the name propagate) because the `Index` has already been created.